### ISSUE TITLE
Custom Loss Functions

### ISSUE THREAD

**Issue by htjb** [htjb/globalemu#4]
Currently the code is restricted to training using the MSE metric (however other metrics are available for testing purposes). It would be good to add additional functionality so that users can pass customised loss functions that are technically more complex or specific to the problem that they are training on.

This can probably be easily implemented by the addition of a keyword argument to the `globalemu.network.nn` code. For example if we call the keyword `loss_function` then the following conditional should suffice

```
if callable(loss_function ):
   loss = loss_function(y, y_)
else:
   loss = default.
```

Needs further exploration and testing however.

**Comment by htjb** [comment]
Closed with #6.
